### ISSUE TITLE
Block tiers

### ISSUE THREAD

**Issue by pyure** [elytra/BetterBoilers#2]
- The version of the mod you are using (e.g. 2.0.0a4)
**1.0**
- Your Minecraft version (e.g. 1.11.2)
**1.12.2**
- Your Forge version (e.g. 13.20.0.2252)
**14.23.0.2549**
- The crash report or log, uploaded to gist.github.com
**N/A**

Similar to the Railcraft boilers, which have two tiers of blocks, would like to see added multiple tiers here (several ideally)

Could probably be done simply with the primary boiler block itself.  Pack devs would then be able to encourage players to upgrade their boilers for higher efficiency/throughput/whatever.

**Comment by LemmaEOF** [comment]
Hm, this might be a bit on the tougher side with how the boiler works. Unlike railcraft, all the boiler components except for vents and valves are dumb blocks; there aren't any TEs attached. The capacity is calculated based off the number of components in the multiblock.

I'm definiely going to have multiple controller types in the near future, but right now the way to get higher throughput is to make a boiler with a bigger footprint, but that comes at the cost of fuel efficiency.

**Comment by LemmaEOF** [comment]
After a long day of coding (and a ton of help from @falkreon), it is now going to be much easier for me to add new boiler controllers! It's likely going to take at least a little more work for me to get full tiering going, and I'm honestly not sure how far I wanna have tiering go, because having new blocks replace your old blocks just isn't that *fun* for me on a fundamental level. I might add an item augment system kinda like Thermal Expansion, or I might add special blocks that add capacity or speed or the like to the boiler. I'll update you here when I decide.

**Comment by pyure** [comment]
Good to hear.  I personally like upgrading individual blocks, especially when the old one can be salvaged in the process, but I respect that that's a subjective preference.  Slotted upgrades would work too.

**Comment by falkreon** [comment]
Yup, and there's plenty of room in the code for, e.g., upgraded (or better yet, specialized) boiler blocks that you can put in the normal boiler multiblock. Needs more discussion and design work.

**Comment by LemmaEOF** [comment]
I've started work on this sorta stuff! Release version 1.1 adds the Boiler Pump, which both has auto-output and makes a boiler work on continuous-flow processing. I'm gonna be adding more blocks like this in the future, including more methods of heating, better boiler wall components, and more. Stay tuned!

**Comment by DrUltraLux** [comment]
I had the idea of implementing Railcraft style pressure, but instead of having a bunch of different tanks, simply have a low pressure and high pressure valve/pump that replace the current pump. 
Simulates the same thing, but the it makes it easier to build modular boiler, and even mix and match the valves at different levels. Could have say a low pressure valve that releases a slow rate of steam as soon as steam starts filling up, and a high pressure that takes over once the steam pressure is high enough.
